src/pages/CouponPage.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="coupon-page">
+    <div class="search-container">
+      <div class="search-core">
+        <input type="text" placeholder="Type here..." class="txt-search" v-model="search" @change="searchMethod">
+        <input type="submit" class="btn-search">
+      </div>
+    </div>
+    <div class="box" v-for="deal in categories" :key="deal.Id">
```

Comment:
```suggestion
    <div class="box" v-for="deal in deals" :key="deal.Id">
```

Variable name must be properly